test/changed.sh

```diff
@@ -17,135 +17,176 @@ set -o errexit
 set -o nounset
 set -o pipefail
 set -o xtrace
+shopt -s nullglob
 
 git remote add k8s https://github.com/kubernetes/charts
 git fetch k8s master
 
-NAMESPACE="pr-${PULL_NUMBER}-${BUILD_NUMBER}"
-CHANGED_FOLDERS=`git diff --find-renames --name-only $(git merge-base k8s/master HEAD) stable/ incubator/ | awk -F/ '{print $1"/"$2}' | uniq`
-CURRENT_RELEASE=""
+release_index=1
+namespace="pr-${PULL_NUMBER}-${BUILD_NUMBER}"
+changed_folders=$(git diff --find-renames --name-only "$(git merge-base k8s/master HEAD)" stable/ incubator/ | awk -F/ '{ print $1"/"$2 }' | uniq)
+current_release=""
 
 # Exit early if no charts have changed
-if [ -z "$CHANGED_FOLDERS" ]; then
-  exit 0
+if [[ -z "$changed_folders" ]]; then
```

Comment:
I noticed you changed the casing on variables in the shell script. If I understand the k8s practice, which I could be wrong about, uppercase variables are for those which are not mutated (essentially constants) and lowercase variables are the ones that change (e.g., in loops). It might be good to follow that convention. Thoughts?

Reply:
I'm all for conventions. In any case the casing was a mixture and not clear as it was. To me, a constant is something that is explicitly initialized as such. It should then probably be made `readonly` as well. The question is whether any variable qualifies as a constant in its actual sense just because it is never changed after it was assigned. According to Google's [Shell Style Guide](https://google.github.io/styleguide/shell.xml?showone=Constants_and_Environment_Variable_Names#Constants_and_Environment_Variable_Names), a constant should be declared at the top of the file. I would agree with that and would change the casing back where this applies.